RecoBTag/SecondaryVertex/plugins/BoostedDoubleSVProducer.cc

```diff
@@ -780,10 +779,47 @@ void BoostedDoubleSVProducer::endStream() {}
 
 // ------------ method fills 'descriptions' with the allowed parameters for the module  ------------
 void BoostedDoubleSVProducer::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
-  //The following says we do not know what parameters are allowed so do no validation
-  // Please change this to state exactly what you do use, even if it is no parameters
   edm::ParameterSetDescription desc;
-  desc.setUnknown();
+  {
+    edm::ParameterSetDescription trackSelection;
+    trackSelection.add<double>("b_pT", 0.3684);
+    trackSelection.add<double>("max_pT", 500);
+    trackSelection.add<bool>("useVariableJTA", false);
+    trackSelection.add<double>("maxDecayLen", 99999.9);
+    trackSelection.add<double>("sip3dValMin", -99999.9);
+    trackSelection.add<double>("max_pT_dRcut", 0.1);
+    trackSelection.add<double>("a_pT", 0.005263);
+    trackSelection.add<unsigned int>("totalHitsMin", 8);
+    trackSelection.add<double>("jetDeltaRMax", 0.3);
+    trackSelection.add<double>("a_dR", -0.001053);
+    trackSelection.add<double>("maxDistToAxis", 0.2);
+    trackSelection.add<double>("ptMin", 1.0);
+    trackSelection.add<std::string>("qualityClass", "any");
+    trackSelection.add<unsigned int>("pixelHitsMin", 2);
+    trackSelection.add<double>("sip2dValMax", 99999.9);
+    trackSelection.add<double>("max_pT_trackPTcut", 3);
+    trackSelection.add<double>("sip2dValMin", -99999.9);
+    trackSelection.add<double>("normChi2Max", 99999.9);
+    trackSelection.add<double>("sip3dValMax", 99999.9);
+    trackSelection.add<double>("sip3dSigMin", -99999.9);
+    trackSelection.add<double>("min_pT", 120);
+    trackSelection.add<double>("min_pT_dRcut", 0.5);
+    trackSelection.add<double>("sip2dSigMax", 99999.9);
+    trackSelection.add<double>("sip3dSigMax", 99999.9);
+    trackSelection.add<double>("sip2dSigMin", -99999.9);
+    trackSelection.add<double>("b_dR", 0.6263);
+    desc.add<edm::ParameterSetDescription>("trackSelection", trackSelection);
+  }
+  desc.add<double>("beta", 1.0);
+  desc.add<double>("R0", 0.8);
+  desc.add<double>("maxSVDeltaRToJet", 0.7);
+  {
+    edm::ParameterSetDescription trackPairV0Filter;
+    trackPairV0Filter.add<double>("k0sMassWindow", 0.03);
+    desc.add<edm::ParameterSetDescription>("trackPairV0Filter", trackPairV0Filter);
+  }
+  desc.add<edm::InputTag>("svTagInfos", edm::InputTag("pfInclusiveSecondaryVertexFinderAK8TagInfos"));
+  desc.add<edm::InputTag>("weights", edm::InputTag(""));
```

Comment:
only the last line was really needed in the context of this PR update.

All other parameter definitions are essentially dead code until all these other configs are actually used in the full module configuration (descriptions.addDefault does not generate an externally usable config).
And even if we do try to use the config (eg with ```descritions.addWithDefaultLabel()``` and then pick up the generated ```boostedDoubleSVProducer_cfi`` in all relevant places) the parts in the trackSelection are a copy trackSelection_cff which are organized to be used in common by many modules, it would have to be injected appropriately without replication.

So, may I suggest to drop the "trackSelection" (the rest could stay perhaps, since it is a bit more specific to this module). And we'll pass a note for BTV developers to implement and use fillDescriptions-based configs.